server/server_setup.go

```diff
@@ -11,14 +11,30 @@ type FHIRServer struct {
 	DatabaseHost     string
 	Engine           *gin.Engine
 	MiddlewareConfig map[string][]gin.HandlerFunc
+	Interceptors     map[string]InterceptorList
 }
 
 func (f *FHIRServer) AddMiddleware(key string, middleware gin.HandlerFunc) {
 	f.MiddlewareConfig[key] = append(f.MiddlewareConfig[key], middleware)
 }
 
+// Adds a new interceptor for a particular HTTP verb and FHIR resource. For example:
+// AddInterceptor("POST", "Patient", handleFunc) would register the function "handleFunc"
+// to be run against a Patient resource when it is created, before being added to the database.
+//
+// To run a handler against ALL resources pass "*" as the resourceType.
+//
+// Supported HTTP verbs are: POST, PUT, DELETE
+func (f *FHIRServer) AddInterceptor(httpVerb, resourceType string, handler InterceptorHandler) {
```

Comment:
If we only support certain ops, consider having this function return an `error` if an unsupported op is passed in.